components/pages/game/components/GameStep.tsx

```diff
@@ -1,92 +1,45 @@
 
-import TimerBar from 'components/shared/atoms/TimerBar';
-import Loader from 'components/shared/atoms/Loader';
+import { PageContent, Content, PreviewImg, ButtonForAnswer, ButtonForQuestion } from './styled';
 
-import { PageContent, Content, PreviewImg, ButtonForAnswer, ButtonForQuestion, TitleDescription } from './styled';
-
-const FULL_BAR = 100;
-const FULL_TIME = 45;
-interface IStep {
-  question: IQuestion | Question;
-  addAnswer: (answer: string) => void;
-}
-
-function useTimer(limit: number, cb = console.log): number {
-  const [time, setTime] = useState(limit);
-  const onceCallRef = useRef(false);
-  if (time <= 0 && onceCallRef.current === false) {
-    cb();
-    onceCallRef.current = true;
-  }
-  useEffect(() => {
-    function tick() {
-      setTime((prevTime) => (prevTime === 0 ? 0 : prevTime - 1));
-    }
-    const id = setInterval(tick, 1000);
-    return () => clearInterval(id);
-  }, [time]);
-  return time;
-}
+type Step = {
+  question: Question;
+  answer: (value: string) => void;
+  isCurrentAnswerCorrect: boolean | null;
+  currentAnswer: string | undefined;
+};
 
-const GameStep = ({ question, addAnswer }: IStep) => {
-  const [selectedId, setSelectedId] = useState<number | null>(null);
-  const isShowResultAnswer = selectedId !== null;
-  const optionsWithUi = question.options.map((name, id) => {
-    const isCorrect = name === question.fullName;
-    const isMatchSelected = id === selectedId;
-    return {
-      id,
-      name,
-      isCorrect,
-      isMatchSelected,
-    };
-  });
-  // const { endGame } = useGameProcess(gameProcess);
-  // const currentSecond = useTimer(FULL_TIME, endGame);
+const GameStep = ({ question, answer, isCurrentAnswerCorrect, currentAnswer }: Step) => {
+  const isShowResultAnswer = Boolean(currentAnswer);
 
-  const barWidth = (FULL_BAR * currentSecond) / FULL_TIME;
-  const callNextStep = (name: string) => () => {
-    setSelectedId(null);
-    addAnswer(name);
-  };
-  if (currentSecond === 0) {
-    return (
-      <Loader testId="profile-updating-loader">
-        <ImageNext src={logoIcon} width={192} height={72} />
-      </Loader>
-    );
-  }
   return (
     <PageContent data-testid="page-content">
-      <TitleDescription>What is the name of that superhero?</TitleDescription>
-      <TimerBar time={currentSecond} width={barWidth} />
       <PreviewImg key={question.avatarUrl} src={question.avatarUrl} />
       <Content>
-        {optionsWithUi.map(({ id, name, isCorrect, isMatchSelected }) => {
-          return isShowResultAnswer ? (
-            <ButtonForAnswer isCorrect={isCorrect} isMatchSelected={isMatchSelected} key={id}>
-              {isCorrect ? <CorrectIcon /> : <IncorrectIcon />}
-              {name}
+        {question.answerOptions.map((option) =>
+          isShowResultAnswer ? (
+            <ButtonForAnswer isCorrect={currentAnswer === option ? isCurrentAnswerCorrect : null} key={option}>
```

Comment:
predicat can be only true or false

Reply:
if we talk about `isCorrect`, it's not predicate, predicate is a function.